pkg/provisioningv2/rke2/planner/probe.go

```diff
@@ -110,9 +110,71 @@ func (p *Planner) addProbes(nodePlan plan.NodePlan, controlPlane *rkev1.RKEContr
 	}
 
 	nodePlan.Probes = replaceRuntimeForProbes(nodePlan.Probes, runtime)
+
+	if isControlPlane(machine) {
+		kcmSecurePort := getArgValue(config[KubeControllerManagerArg], SecurePortArgument, "=")
+		if kcmSecurePort == "" {
+			kcmSecurePort = KubeControllerManagerDefaultSecurePort
+		}
+		kcmTLSCert := getArgValue(config[KubeControllerManagerArg], TLSCertFileArgument, "=")
+		if kcmTLSCert == "" {
+			kcmCertDir := getArgValue(config[KubeControllerManagerArg], CertDirArgument, "=")
+			if kcmCertDir == "" {
+				kcmCertDir = fmt.Sprintf(KubeControllerManagerCertDir, rancherruntime.GetRuntime(controlPlane.Spec.KubernetesVersion))
+			}
+			// we can use the kube-controller-manager cert-dir value and port
+			kcmProbe, err := replaceCACertAndPortForProbes(nodePlan.Probes["kube-controller-manager"], kcmCertDir+"/kube-controller-manager.crt", kcmSecurePort)
+			if err != nil {
+				return nodePlan, err
+			}
+			nodePlan.Probes["kube-controller-manager"] = kcmProbe
+		} else {
+			// We need to use the kube-controller-manager TLS Cert and Port
+			kcmProbe, err := replaceCACertAndPortForProbes(nodePlan.Probes["kube-controller-manager"], kcmTLSCert, kcmSecurePort)
+			if err != nil {
+				return nodePlan, err
+			}
+			nodePlan.Probes["kube-controller-manager"] = kcmProbe
+		}
```

Comment:
Another simplification since the end of the if and else parts were the same: track the value of `kcmTLSCert` and append the probe at the end.
```suggestion
		kcmSecurePort := getArgValue(config[KubeControllerManagerArg], SecurePortArgument, "=")
		if kcmSecurePort == "" {
			kcmSecurePort = KubeControllerManagerDefaultSecurePort
		}
		kcmTLSCert := getArgValue(config[KubeControllerManagerArg], TLSCertFileArgument, "=")
		if kcmTLSCert == "" {
			kcmCertDir := getArgValue(config[KubeControllerManagerArg], CertDirArgument, "=")
			if kcmCertDir == "" {
				kcmCertDir = fmt.Sprintf(KubeControllerManagerCertDir, rancherruntime.GetRuntime(controlPlane.Spec.KubernetesVersion))
			}
			kcmTLSCert = kcmCertDir+"/kube-controller-manager.crt"
		}
		 kcmProbe, err := replaceCACertAndPortForProbes(nodePlan.Probes["kube-controller-manager"], kcmTLSCert, kcmSecurePort)
		 if err != nil {
			 return nodePlan, err
		 }
		 nodePlan.Probes["kube-controller-manager"] = kcmProbe
```

Reply:
Simplified